### DISCUSSION TITLE
How to download Audit Trails?

### DISCUSSION THREAD

**Discussion by YMov** [Hugovdberg/PIconnect#671] [Q&A]
Hi, 

I want to download AF Audit Trails using PIconnect rather than manually downloading it in PI PSE. Does PIconnect have this option? 

Thanks,
Yazdan

**Comment by Hugovdberg** [comment]
Currently I don't think it is possible using the python API, although it should be possible to implement using the Osisoft AF SDK. I don't have time to investigate this the next few weeks, but the entire OSisoft API is available from `PIconnect.AFSDK.AF`.
